addons/sale_project/models/project.py

```diff
@@ -356,20 +357,30 @@ def _get_profitability_sale_order_items_domain(self, domain=None):
     def _get_revenues_items_from_sol(self, domain=None, with_action=True):
         sale_line_read_group = self.env['sale.order.line'].sudo()._read_group(
             self._get_profitability_sale_order_items_domain(domain),
-            ['product_id', 'ids:array_agg(id)', 'untaxed_amount_to_invoice', 'untaxed_amount_invoiced'],
-            ['product_id'],
+            ['product_id', 'ids:array_agg(id)', 'untaxed_amount_to_invoice', 'untaxed_amount_invoiced', 'currency_id'],
+            ['product_id', 'currency_id'],
+            lazy=False,
         )
         display_sol_action = with_action and len(self) == 1 and self.user_has_groups('sales_team.group_sale_salesman')
         revenues_dict = {}
         total_to_invoice = total_invoiced = 0.0
         if sale_line_read_group:
-            sols_per_product = {
-                res['product_id'][0]: (
-                    res['untaxed_amount_to_invoice'],
-                    res['untaxed_amount_invoiced'],
-                    res['ids'],
-                ) for res in sale_line_read_group
-            }
+            # Get conversion rate from currencies to currency of analytic account
+            currency_ids = list(set([line['currency_id'][0] for line in sale_line_read_group] + [self.analytic_account_id.currency_id.id]))
+            rates = self.env['res.currency'].browse(currency_ids)._get_rates(self.company_id, date.today())
+            conversion_rates = {cid: rates[self.analytic_account_id.currency_id.id] / rate_from for cid, rate_from in rates.items()}
+
+            sols_per_product = {}
+            for group in sale_line_read_group:
+                product_id = group['product_id'][0]
+                currency_id = group['currency_id'][0]
+                sols_total_amounts = sols_per_product.setdefault(product_id, (0, 0, []))
+                sols_current_amounts = (
+                    group['untaxed_amount_to_invoice'] * conversion_rates[currency_id],
+                    group['untaxed_amount_invoiced'] * conversion_rates[currency_id],
+                    group['ids'],
+                )
+                sols_per_product[product_id] = tuple(reduce(lambda x, y: x + y, pair) for pair in zip(sols_total_amounts, sols_current_amounts))
```

Comment:
I think we should use the currency of the project instead of using the one in AA linked to the project. Moreover, the currency in the AA could be empty since the company is not required. And also, in JS code, we will use the currency of the project to format the amount. 🙂

Reply:
Ok, I wasn't sure which one to use, but since the fix 68b4ea6 uses the AA's currency, I followed the same for consistency. But since the front-end will use the project's currency. I will update all instances of currency conversion to use the project's instead :)